xwiki-platform-core/xwiki-platform-comment/xwiki-platform-comment-api/src/main/java/org/xwiki/platform/comment/CommentManager.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.platform.comment;
+
+import java.util.List;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.model.reference.EntityReference;
+
+@Role
+public interface CommentManager
+{
+    List<Comment> getComments(EntityReference documentReference, boolean ascendindOrder) throws CommentException;
```

Comment:
I guess we might need to return more something like a `CommentTree` if we plan to support threads in that API.

We might want something a bit more advanced in terms of input to cover stuff like pagination in the future.

Reply:
Hmmm I was initially thinking about providing later another API to get the comments with the threads structure. Do you think we should have only a single one?

Reply:
> Do you think we should have only a single one?

I don't see much use case that would not want the tree. But even if there is one, IMO the simplest in terms of API would probably be more to consider giving the comments as a tree or a list as features of the `CommentTree`.